### ISSUE TITLE
Migrating from Redis to Faktory 

### ISSUE THREAD

**Issue by j-mcnally** [akira/exq#291]
The author of Sidekiq (@mperham) has released Faktory
http://www.mikeperham.com/2017/10/24/introducing-faktory/

The TLDR; is that Faktory provides a Web UI, and a job queue / management server.

It is then the responsibility of an author to code a client, and worker (process) manager.

This allows any language to work with Faktory jobs, and for a single team to use a single Faktory server and write jobs in any language. It also replaces redis with its own wrapped implementation of RocksDB and a smart server purpose built for managing jobs.

We would just need to write a client in Elixir to push jobs to Faktory, and then some sort of code (supervisor / manager) to dequeue a job, spin up a new process and run the appropriate worker.

I know the goal of Exq, atleast originally was to be sidekiq compatible, but it seems as if the eco-system and contribsys are moving away from multi-langague compatible sidekiq, to a more standardized polyglot architecture with Faktory. 

I believe this would be a good change for Exq / Elixir to join this eco system, and I would love thoughts.

**Comment by mperham** [comment]
I appreciate the gusto but it's still really, really early days for Faktory.

If you are interested in using Faktory from Elixir, check out [[https://github.com/cjbottaro/faktory_worker_ex]].  I bet Chris would appreciate some help.

**Comment by akira** [comment]
@mperham thanks for chiming in and the link to the library! (as well as your numerous excellent open source work). 

@j-mcnally I agree that this will be a likely direction that the eco system will move towards. This is a clear advantage versus having to re-implement all this logic in every language. However, I think this would be good to tackle this as a new repo with a fresh approach not tied to the Redis based implementation which needs to handle a lot more concerns. Also, going forward, people will still be using Sidekiq for their legacy services even as they are migrating off to Faktory so I don't necessarily want to remove that functionality.

**Comment by bschaeffer** [comment]
👎 on switching. Better solution would be swap-able backends.

**Comment by davidkovsky** [comment]
👍 to swappable backends.
